cmd/kubeadm/app/util/patches/patches_test.go

```diff
@@ -172,8 +173,8 @@ func TestGetPatchSetsForPathMustBeDirectory(t *testing.T) {
 	defer os.Remove(tempFile.Name())
 
 	_, _, _, err = getPatchSetsFromPath(tempFile.Name(), testKnownTargets, ioutil.Discard)
-	if err == nil {
-		t.Fatalf("expected error for non-directory path %q", tempFile.Name())
+	if !strings.Contains(err.Error(), "not a directory") {
+		t.Fatalf("expected \"not a directory\" error for non-directory path %q", tempFile.Name())
```

Comment:
error checks with string matching against go stdlib are not ideal.
AFAIK, there is no OS agnostic NOTADIRERR that can be checked here, so `err == nil` is better even if not completely accurate.

Reply:
@neolit123 Thank you for the review. But this is not matching against go stdlib. It is this line: https://github.com/kubernetes/kubernetes/blob/6a667de8d8ec8c58e32b8716f2a5c748b285985f/cmd/kubeadm/app/util/patches/patches.go#L286

Reply:
Ok my mistake. In this case you can make it return a typed error and check the type. This is still better than string matching.

/reopen

Reply:
@neolit123 I updated the PR. Let me know if I need to make further changes.